src/solvers/refinement/bv_refinement_loop.cpp

```diff
@@ -123,6 +123,10 @@ void bv_refinementt::check_SAT()
 
   arrays_overapproximated();
 
+  // get values before modifying the formula
```

Comment:
How did you notice this? Is there a test for that or does it fail quite broadly?

Reply:
I actually noticed this while working on #4758, because Cadical complained. For the IPASIR interface this commit was just an attempt to make things work, but we still have failing regression tests, so I need to dig deeper.